hs_core/views/serializers.py

```diff
@@ -0,0 +1,75 @@
+    edit_groups = serializers.MultipleChoiceField(choices=group_choices, required=False)
+    view_users = serializers.MultipleChoiceField(choices=user_choices, required=False)
+    view_groups = serializers.MultipleChoiceField(choices=group_choices, required=False)
+
+
+class ResourceCreateRequestValidator(ResourceUpdateRequestValidator):
+    resource_type = serializers.ChoiceField(
+            choices=zip(
+                [x.__name__ for x in hydroshare.get_resource_types()],
+                [x.__name__ for x in hydroshare.get_resource_types()]
+            ), default='GenericResource')
+
+
+class ResourceListRequestValidator(serializers.Serializer):
+    group = serializers.CharField(min_length=1, required=False, validators=[validate_group_name])
+    user = serializers.CharField(min_length=1, required=False, validators=[validate_user_name])
+    owner = serializers.CharField(min_length=1, required=False, validators=[validate_user_name])
+    from_date = serializers.DateField(required=False, default=None)
+    to_date = serializers.DateField(required=False, default=None)
+    start = serializers.IntegerField(required=False, default=None)
+    count = serializers.IntegerField(required=False, default=None)
+    metadata = serializers.CharField(min_length=1, required=False, validators=[validate_json])
+    full_text_search = serializers.CharField(required=False)
+    edit_permission = serializers.BooleanField(required=False, default=False)
+    published = serializers.BooleanField(required=False, default=False)
+    resource_types = utils.get_resource_types()
+    resource_type_choices = [rtype.__name__ for rtype in resource_types]
+    types = serializers.MultipleChoiceField(choices=resource_type_choices, required=False, default=None)
+
+
+class ResourceListItemSerializer(serializers.Serializer):
+    resource_type = serializers.CharField(max_length=100)
+    resource_title = serializers.CharField(max_length=200)
+    resource_id = serializers.CharField(max_length=100)
+    creator = serializers.CharField(max_length=100)
+    date_created = serializers.DateTimeField(format='%d-%m-%Y')
+    date_last_updated = serializers.DateTimeField(format='%d-%m-%Y')
+    sharing_status = serializers.CharField(max_length=20)
+    bag = serializers.URLField()
+    science_metadata = serializers.URLField()
+
+
+class ResourceListItem(object):
+    def __init__(self, resource_type, resource_id, resource_title, creator_name, sharing_status, date_created,
+                 date_last_updated, bag_url, science_metadata_url):
+        self.resource_type = resource_type
+        self.resource_id = resource_id
+        self.resource_title = resource_title
+        self.creator = creator_name
+        self.sharing_status = sharing_status
+        self.date_created = date_created
+        self.date_last_updated = date_last_updated
+        self.bag = bag_url
+        self.science_metadata = science_metadata_url
```

Comment:
might be easier to use a namedtuple here. will make it easier to edit in the future, at least.